app/utils/JsResultSupport.scala

```diff
@@ -0,0 +1,12 @@
+package utils
+
+import play.api.libs.json.JsResult
+
+object JsResultSupport {
+  def map2[A, B, C](resultA: JsResult[A], resultB: JsResult[B])(f: (A, B) => C): JsResult[C] =
```

Comment:
I understand what's happening here but a short explanation might be useful. Also, it may be more self explanatory to call the method `combineJsResults`/`transformJsResults` or something similar.

Reply:
Whilst it may seem weird at first, map2 is actually the common name for this operation.

map applies a function of one argument to the encapsulated value.
map2 applies a 2 argument function - and therefore requires 2 instances of the type.